### PR TITLE
CIVIPLUS-180: Simplify afform event naming

### PR DESCRIPTION
**Pull request by ahed-compucorp**
Overview
----------------------------------------
This PR fixes the missing class error that stopped the deployment for sites with `afform` extensions enabled.

Before
----------------------------------------
The deployment process will fail with the following error:
```php
Error: Class "Civi\Api4\Action\Afform\Submit" not found in afform_civicrm_config() (line 52 of civicrm/ext/afform/core/afform.php).
Drush command terminated abnormally due to an unrecoverable error
```

After
----------------------------------------
The deployment process will finish successfully.

Technical Details
----------------------------------------
The deployment process complained about the missing class `Civi\Api4\Action\Afform\Submit`. I investigated the issue and confirmed the file was loaded: 
```bash
# Added 'var_dump(Submit::class);die()' before afform.php:52
$ drush rr                                                                                     
The registry has been rebuilt via registry_rebuild (A).                                                                                                       [success]
"continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? advuser.module:552                                                    [warning]
string(30) "Civi\Api4\Action\Afform\Submit"
Drush command terminated abnormally due to an unrecoverable error.
```

To overcome this issue, I used the sting literal instead of the constant for the event name.
```diff
-  Civi::dispatcher()->addListener(Submit::EVENT_NAME, [Submit::class, 'processContacts'], 500);                                                                         
-  Civi::dispatcher()->addListener(Submit::EVENT_NAME, [Submit::class, 'processGenericEntity'], -1000);                                                                  
+  Civi::dispatcher()->addListener('civi.afform.submit', ['\Civi\Api4\Action\Afform\Submit', 'processContacts'], 500);                                                   
+  Civi::dispatcher()->addListener('civi.afform.submit', ['\Civi\Api4\Action\Afform\Submit', 'processGenericEntity'], -1000);  
```

Since CiviCRM 5.26, the convention was to use string literal for event names and the usage of constants was deprecated according to this issue [#1744](https://lab.civicrm.org/dev/core/-/issues/1744). This was the last usage of constants for event names in CiviCRM code base.

Comments
----------------------------------------
I couldn't replicate this issue in any of my local sites.

### CODE REVIEW

File: ext/afform/core/afform.php
Comment (erawat) [inline ext/afform/core/afform.php:51]: @ahed-compucorp  I am wondering if we should leave Submit::EVENT_NAME as it is as the issue is the class name that does not include name space?

Comment (ahed-compucorp) [reply ext/afform/core/afform.php:51]: I tried to leave `Submit::EVENT_NAME`, but the same error was thrown again, but it works if I left `Submit::class`.

Comment (erawat) [reply ext/afform/core/afform.php:51]: Very interesting.   I'll leave this to you to decide and see what core team will decide.

**Review by erawat — APPROVED** [review]

**Review by erawat — APPROVED** [review]

**Pull request merged**
